### ISSUE TITLE
Questions for Hayden

### ISSUE THREAD

**Issue by ArturoAmaya** [ArturoAmaya/ExploratoryCurricularAnalytics#40]
A thread with questions to ask the CA team

**Comment by ArturoAmaya** [comment]
### Do the course IDs have to be sequential?
That is, can a course on line `7` have course ID `4` when the one below it has course ID `10`?

**Comment by ArturoAmaya** [comment]
### How important is the co-requisite information?
Per the CA documentation
>Co-requisite : course *A* may be taken prior to or at the same time as attempting course *B*

That means that courses like `CSE 30` and `CSE 105` are co-requisites of each other - they both depend on `CSE 12` and `CSE 15L` (`CSE 105` also depends on `CSE 21`) and there's no problem taking one of them before or at the same time as the other. This happens *a lot*, especially if we have a bunch of generic elective classes. Two issues arise:

- It seems to me that this muddies up the graph without adding too much information. It also begs a choice of, for any two pairs of co-requisite courses, which one gets the outgoing arrow and why?
- We haven't developed a reliable method of identifying co-requisites (other than not-pre-requisites)

**Comment by ArturoAmaya** [comment]
That's true, I haven't seen any option for that.

**Comment by ArturoAmaya** [comment]
### Can we move the little stats pop-up?
It's so minor, but going through the MC25 curriculum there's a few cases like this:
![image](https://user-images.githubusercontent.com/47407486/174398342-6d742c37-31a2-4b07-a66c-68ee8b72809a.png)
`MAE 156` seems to need `MAE 131A` and `MAE 30B` and `MAE 170`. It also seems to be missing a pre-req: `MAE 150`. It's just that  `MAE 150` is on there, just hidden by the stats pop-up (which sometimes, but not always becomes translucent).
![image](https://user-images.githubusercontent.com/47407486/174398736-6316c98d-d141-41a2-a8d7-7e45346c93e2.png)
Is there any way to move it to be above the node for a given course, or to one side?

**Comment by ArturoAmaya** [comment]
### Is there someway to move the little arrows?
Also super minor, but it got annoying after like 20 times. The requisite relation arrows sometimes go precisely over the name of a course. Can we just make the arc of those different?
![image](https://user-images.githubusercontent.com/47407486/174399115-374b7469-88ea-42a8-bbbf-de15a91ce50d.png)
Here, `MATH 20D` is obscured by the arrow from `MATH 20C` to `MATH 20E`. When they appear together in a pre/co-requisite field, like for `MAE 101A` the fact that the longest path includes the mentioned arrow makes is thicker and blue. It makes `MATH 20D` annoying to read. It's *super minor*, I know, but it really started bugging me after a while.

**Comment by SheepTester** [comment]
> ### Can we move the little stats pop-up?

If it doesn't become translucent I think you can hover over it again and hope it becomes translucent
![overwrite me](https://user-images.githubusercontent.com/22133785/174399949-147de9b5-87d1-42c0-bda4-878b9d929efb.gif)

I don't think you can move the element (you could get rid of it using a browser extension perhaps), so I think this is a UI issue. We might be working on the UI after this later in the summer, I think?

**Comment by SheepTester** [comment]
> it'd also be nice to know if we can reupload CSVs for curricula/degree plans since if we uploaded all the major plans and then realize we made a little oopsie it would be quite catastrophic if we couldn't change them

![image](https://user-images.githubusercontent.com/22133785/174403032-10613292-5cf2-4571-9055-e5f9215f90f8.png)
As it turns out, we can destroy curricula/degree plans we uploaded, so if we need to change something we at least have the option to delete and recreate it

**Comment by ArturoAmaya** [comment]
Yeah but when you destroy you have to create the name and the org and so on... it'd be nice to just be able to update the document used. 🤷 Oh yeah and about the translucent thing it sometimes doesn't ever become translucent even if I rehover I have no clue why 😢

**Comment by ArturoAmaya** [comment]
### For analysis purposes, does it matter if some DEIs are listed in the curriculum section of the degree plans document?
It seems that visually, there's no difference or complaints from Curricular Analytics. Example with AN26, Warren and ERC.
![image](https://user-images.githubusercontent.com/47407486/174647991-3004bd1b-270f-4458-bff2-dcbb2f9ebf5c.png)
![image](https://user-images.githubusercontent.com/47407486/174648097-ccce5ea2-1ab4-4a28-9e68-7db4fd93268e.png)
The DEI course being in different places isn't a problem it seems. 
However the files don't match up:
![image](https://user-images.githubusercontent.com/47407486/174648332-8cde70ee-c423-4f05-a9e1-5fce4a064068.png)
When we're analyzing everything does that make a difference? I'm hesitant to call it a bug because it seems to happen:
- Rarely 
- Only with DEIs, which in our abstract sense don't have pre-reqs or unblocked classes
- Only in majors where the classes are taken from a list, i.e. the ones with courses called "UD Elective" or something like that, which also have no pre-reqs or unblocked classes, so the fact that the DEI occupies the same space as another class in the list seems to not affect anything.

Todo on this:
- [ ] Find a major where this happens where the DEI displaces a course with links to other courses. (that would level this up to bug)

- [ ] Consider making a list of majors where DEIs can overlap with the major and standardize across college degree plans (that is, if a DEI can overlap with a major class, it should do so regardless of college. Then again, the we're supposed to follow the degree plans, no?)

**Comment by SheepTester** [comment]
![image](https://user-images.githubusercontent.com/22133785/174655791-3b88ed2a-fa92-41c1-9904-e7f667358083.png)
That's weird, in academic_plans.csv they say there's overlap for DEI between college and department

**Comment by ArturoAmaya** [comment]
Yup, some colleges are weird about that. Warren specifically does that a lot. I'll type up what I've found when I get back from lunch. 😀

**Comment by SheepTester** [comment]
I wonder what these fields in the header are used for in the CSV files
![image](https://user-images.githubusercontent.com/22133785/175410845-6679e69b-3fb7-45fa-bc3b-13089cd7bf7a.png)
It seems that Curricular Analytics only cares about the CIP code and the other fields are ignored
